src/contracts/libraries/BN254.sol

```diff
@@ -139,19 +139,40 @@ library BN254 {
      */ 
     function scalar_mul_tiny(BN254.G1Point memory p, uint16 s) internal view returns (BN254.G1Point memory) {
         require(s < 2**9, "scalar-too-large");
+
         // the accumulated product to return
         BN254.G1Point memory acc = BN254.G1Point(0, 0);
-        // the 2^n*p to add to the accumulated product in each iteration
-        BN254.G1Point memory p2n = p;
-        // loop through each bit of s
-        for (uint8 i = 0; i < 9; i++) {
-            // if the bit is 1, add the 2^n*p to the accumulated product
-            if (s >> i & 1 == 1) {
-                acc = plus(acc, p2n);
+        // value of most signifigant bit
+        uint16 m = 1;
+
+        //check if s is greater than 1
+        if(s > m){
```

Comment:
this function returns the wrong thing if s == 0, probably, just add a conditional at the top if s == 1